themes/tfjs/source/js/codeSnippets.js

```diff
@@ -157,19 +157,26 @@ function initCodeBlocks(selector) {
 
     block.parentElement.insertAdjacentElement('afterend', consoleElement);
 
-    consoleRunElement.addEventListener('click', function() {
+    function runCodeSnippet() {
       var consoleLogElement =
-        this.parentElement.querySelector('.snippet-console-log');
+          this.parentElement.querySelector('.snippet-console-log');
 
       var snippetText;
       if (block.codeMirror) {
         snippetText = block.codeMirror.getValue();
       } else {
         snippetText = this.parentElement.previousElementSibling.innerText;
       }
-
       executeCodeSnippet(consoleLogElement, snippetText);
-    });
+    }
+
+    if (tf != null && tf.ready != null) {
+      tf.ready().then(() => {
+        consoleRunElement.addEventListener('click', runCodeSnippet);
+      });
+    } else {
+      consoleRunElement.addEventListener('click', runCodeSnippet);
+    }
```

Comment:
If `tf` is sometimes null, then maybe it's better to check for it each time the user clicks on `run`. We can make `runCondeSnippet` async and have it check for `tf`. If it's available, we can await `tf.ready()` to make sure it's set up. Otherwise, we can show a warning message.

What do you think?

Reply:
This should also catch any future cases where we're switching backends and the new backend isn't initialized yet. This situation won't appear now, but if we add a backend dropdown, then it could happen.

Reply:
Nice catch, thanks! Done.